src/blame.rs

```diff
@@ -0,0 +1,167 @@
+use std::marker;
+use {raw, Repository, Oid, signature, Signature};
+use util::{self, Binding};
+use std::path::Path;
+use std::ops::Range;
+
+/// Opaque structure to hold blame results.
+pub struct Blame<'repo> {
+    raw: *mut raw::git_blame,
+    _marker: marker::PhantomData<&'repo Repository>,
+}
+
+/// Structure that represents a blame hunk.
+pub struct BlameHunk<'blame> {
+    raw: *mut raw::git_blame_hunk,
+    _marker: marker::PhantomData<&'blame raw::git_blame>,
+}
+
+/// An iterator over the hunks in a blame.
+pub struct BlameIter<'blame> {
+    range: Range<usize>,
+    blame: &'blame Blame<'blame>,
+}
+
+impl<'repo> Blame<'repo> {
+
+    /// Gets the number of hunks that exist in the blame structure.
+    pub fn len(&self) -> usize {
+        unsafe { raw::git_blame_get_hunk_count(self.raw) as usize }
+    }
+
+    /// Gets the blame hunk at the given index.
+    pub fn get_index(&self, index: u32) -> Option<BlameHunk> {
```

Comment:
Although the underlying API takes `u32`, this may be best to take a `usize` to be consistent with `len`